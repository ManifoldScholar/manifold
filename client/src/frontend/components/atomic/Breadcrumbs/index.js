import React from "react";
import PropTypes from "prop-types";
import IconComposer from "global/components/utility/IconComposer";
import * as Styled from "./styles";

export default function Breadcrumbs({ breadcrumbs, secondary = false }) {
  const OuterComponent = secondary ? Styled.OuterSecondary : Styled.Outer;
  const InnerComponent = secondary ? Styled.InnerSecondary : Styled.Inner;
  const icon = secondary ? "circleArrowLeft64" : "arrowLeft16";
  const iconSize = secondary ? 56 : 24;

  return (
    <OuterComponent className="bg-neutral05">
      <InnerComponent>
        <IconComposer icon={icon} size={iconSize} />
        {breadcrumbs.map(crumb => (
          <Styled.Breadcrumb to={crumb.to} key={crumb.to}>
            {crumb.label}
          </Styled.Breadcrumb>
        ))}
      </InnerComponent>
    </OuterComponent>
  );
}

Breadcrumbs.displayName = "Frontend.Atomic.Breadcrumbs";

Breadcrumbs.propTypes = {
  breadcrumbs: PropTypes.object.isRequired,
  secondary: PropTypes.bool
};
