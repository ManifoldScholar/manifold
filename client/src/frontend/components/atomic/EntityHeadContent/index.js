import React from "react";
import PropTypes from "prop-types";
import HeadContent from "global/components/HeadContent";
import { useSelectSettings } from "hooks";

function EntityHeadContent({ entity, parentEntity = { attributes: {} } }) {
  const settings = useSelectSettings();

  const installationName = settings?.attributes.general.installationName || "";
  const {
    socialTitle,
    socialDescription,
    socialImageStyles,
    titlePlaintext,
    descriptionPlaintext,
    heroStyles
  } = entity.attributes;
  const {
    socialDescription: parentSocialDescription,
    socialImageStyles: parentSocialImageStyles,
    titlePlaintext: parentTitle,
    descriptionPlaintext: parentDescription,
    heroStyles: parentHeroStyles
  } = parentEntity?.attributes;

  const title = () => {
    if (socialTitle) return socialTitle;
    const title = parentTitle
      ? `${parentTitle}: ${titlePlaintext}`
      : titlePlaintext;
    return `\u201c${title}\u201d on ${installationName}`;
  };

  const description = () => {
    if (socialDescription) return socialDescription;
    if (descriptionPlaintext) return descriptionPlaintext;
    if (parentSocialDescription) return parentSocialDescription;
    if (parentDescription) return parentDescription;
    return null;
  };

  const image = () => {
    if (socialImageStyles?.mediumLandscape)
      return socialImageStyles.mediumLandscape;
    if (heroStyles?.mediumLandscape) return heroStyles.mediumLandscape;
    if (parentSocialImageStyles?.mediumLandscape)
      return parentSocialImageStyles.mediumLandscape;
    if (parentHeroStyles?.mediumLandscape)
      return parentHeroStyles.mediumLandscape;
    return null;
  };

  return (
    <HeadContent title={title()} description={description()} image={image()} />
  );
}

EntityHeadContent.propTypes = {
  entity: PropTypes.object.isRequired,
  parentEntity: PropTypes.object
};

export default EntityHeadContent;
