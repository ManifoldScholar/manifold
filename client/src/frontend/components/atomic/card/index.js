import React from "react";
import { Link } from "react-router-dom";
import lh from "helpers/linkHandler";
import Utility from "global/components/utility";
import PropTypes from "prop-types";
import classNames from "classnames";

const Card = ({
  header = false,
  title,
  link = [],
  linkStyle = { textDecoration: "none" },
  icon,
  iconSize = 24,
  blockClass = "frontend",
  children
}) => {
  const headerContent = () => {
    const linked = link.length > 0;
    const getClass = base => {
      return classNames({
        [base]: true,
        "is-linked": linked
      });
    };
    const content = (
      <div className={getClass("card__header")}>
        <div className="card__header-title">{title}</div>
        {icon && (
          <div className={getClass("card__header-icon")}>
            <Utility.IconComposer icon={icon} size={iconSize} />
          </div>
        )}
      </div>
    );
    return link.length > 0 ? (
      <Link style={linkStyle} to={lh.link(...link)}>
        {content}
      </Link>
    ) : (
      <div>{content}</div>
    );
  };
  return (
    <div className={blockClass}>
      <div className="card">
        {header && <div>{headerContent()}</div>}
        <div className="card__body">{children}</div>
      </div>
    </div>
  );
};

Card.propTypes = {
  header: PropTypes.bool,
  title: PropTypes.string,
  link: PropTypes.array,
  linkStyle: PropTypes.object,
  icon: PropTypes.string,
  blockClass: PropTypes.string
};

export default Card;
