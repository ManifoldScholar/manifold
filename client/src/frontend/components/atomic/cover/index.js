import React from "react";
import PropTypes from "prop-types";
import UniqueIcons from "global/components/icon/unique";
import Image from "global/components/atomic/Image";
import classNames from "classnames";

const Cover = ({ entity, blockClass = "project-hero" }) => {
  const cover = () => {
    const { coverStyles, avatarMeta } = entity.attributes;
    if (!coverStyles || !avatarMeta) return;
    return {
      url: entity.attributes.coverStyles.medium,
      width: "100%",
      height: "auto"
    };
  };

  const renderAvatarImage = () => {
    if (!entity.attributes.avatarMeta.original) return null;
    const meta = entity.attributes.avatarMeta.original;
    const imageStyle =
      meta.width >= meta.height
        ? entity.attributes.avatarStyles.smallSquare
        : entity.attributes.avatarStyles.small;
    return <Image image={imageStyle} title="Cover" />;
  };

  const renderPlaceholderImage = () => {
    if (!entity.attributes.avatarColor) return null;
    return (
      <UniqueIcons.ProjectPlaceholderUnique
        mode="responsive"
        color={entity.attributes.avatarColor}
        ariaLabel={false}
      />
    );
  };

  const figureClassNames = classNames(`${blockClass}__figure`, {
    [`${blockClass}__figure--constrained`]: !cover().url
  });

  return (
    <div className={`${blockClass}__cover-block`}>
      <figure className={figureClassNames}>
        {/* eslint-disable-next-line no-nested-ternary */}
        {cover() ? (
          <Image image={cover()} title="Cover" />
        ) : entity.attributes.avatarStyles.original ? (
          renderAvatarImage()
        ) : (
          renderPlaceholderImage()
        )}
      </figure>
    </div>
  );
};

Cover.displayName = "Entity.Cover";

Cover.propTypes = {
  blockClass: PropTypes.string,
  entity: PropTypes.object.isRequired
};

export default Cover;
