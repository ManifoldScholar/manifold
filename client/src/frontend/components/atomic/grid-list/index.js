import React from "react";
import PropTypes from "prop-types";
import Utility from "global/components/utility";
import { TransitionGroup as ReactTransitionGroup } from "react-transition-group";
import { Link } from "react-router-dom";
import classNames from "classnames";

const GridList = ({
  children,
  limit = 0,
  pagination,
  paginationTarget,
  paginationClickHandler,
  showViewAll,
  viewAllUrl,
  viewAllLabel = "See All Projects"
}) => {
  const itemsList = () => {
    if (limit === 0) return [];
    return limit
      ? React.Children.toArray(children).slice(0, limit)
      : React.Children.toArray(children);
  };

  const viewAll = () => {
    return (
      showViewAll ||
      (limit && viewAllUrl && React.Children.count(children) > limit)
    );
  };

  const renderViewAll = () => {
    if (!viewAll()) return null;

    return (
      <div
        className={classNames(
          "entity-section-wrapper__utility",
          "entity-section-wrapper__utility--footer"
        )}
      >
        <Link to={viewAllUrl}>
          {viewAllLabel}
          <Utility.IconComposer icon="arrowLongRight16" size="default" />
        </Link>
      </div>
    );
  };

  const renderPagination = () => {
    if (pagination?.totalPages <= 1) return null;
    return (
      <div className="entity-section-wrapper__pagination">
        <Utility.Pagination
          paginationClickHandler={paginationClickHandler}
          pagination={pagination}
          paginationTarget={paginationTarget}
        />
      </div>
    );
  };

  const items = itemsList();

  if (items.length === 0) return null;
  return (
    <>
      <div className="grid-list grid entity-section-wrapper__body">
        <ReactTransitionGroup component="ul">{items}</ReactTransitionGroup>
      </div>
      {pagination ? renderPagination() : renderViewAll()}
    </>
  );
};

GridList.displayName = "Atomic.GridList";

GridList.propTypes = {
  children: PropTypes.node,
  limit: PropTypes.number,
  pagination: PropTypes.object,
  paginationTarget: PropTypes.string,
  paginationClickHandler: PropTypes.func,
  showViewAll: PropTypes.bool,
  viewAllUrl: PropTypes.string,
  viewAllLabel: PropTypes.string
};

export default GridList;
