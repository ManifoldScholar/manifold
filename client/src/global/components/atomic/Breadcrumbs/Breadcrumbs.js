import React from "react";
import PropTypes from "prop-types";
import Frontend from "./Frontend";
import Backend from "./Backend";

export default function Breadcrumbs({ backend, ...props }) {
  return backend ? <Backend {...props} /> : <Frontend {...props} />;
}

Breadcrumbs.displayName = "Global.Atomic.Breadcrumbs";

Breadcrumbs.propTypes = {
  breadcrumbs: PropTypes.array.isRequired,
  backend: PropTypes.bool,
  hideOnDesktop: PropTypes.bool
};
