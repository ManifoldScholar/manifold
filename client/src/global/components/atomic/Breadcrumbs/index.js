export { default } from "./Breadcrumbs";

export {
  default as BreadcrumbsProvider,
  BreadcrumbsContext,
  RegisterBreadcrumbs
} from "./Context";
