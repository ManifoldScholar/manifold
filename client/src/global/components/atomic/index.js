export { default as HeaderLogo } from "./HeaderLogo";
