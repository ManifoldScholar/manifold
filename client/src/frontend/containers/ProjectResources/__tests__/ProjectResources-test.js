import ProjectResourcesContainer from "../";
import { BreadcrumbsProvider } from "global/components/atomic/Breadcrumbs";
import useSetLocation from "hooks/useSetLocation";

jest.mock("hooks/useSetLocation", () => () => null);

describe("frontend/containers/ProjectResources/ProjectResources", () => {
  def("project", () => factory("project"));
  def("root", () => (
    <BreadcrumbsProvider>
      <ProjectResourcesContainer
        project={$project}
      />
    </BreadcrumbsProvider>
  ));
  it("matches the snapshot", () => {
    expect(mount($withApp($root))).toMatchSnapshot();
  });
});
