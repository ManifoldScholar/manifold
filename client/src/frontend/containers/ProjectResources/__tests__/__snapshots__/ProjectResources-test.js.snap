// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`frontend/containers/ProjectResources/ProjectResources matches the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
